src/extension/providers/nostr/index.ts

```diff
@@ -10,16 +12,23 @@ declare global {
 export default class NostrProvider {
   nip04 = new Nip04(this);
   enabled: boolean;
+  private _eventEmitter: EventEmitter;
 
   constructor() {
     this.enabled = false;
+    this._eventEmitter = new EventEmitter();
   }
 
   async enable() {
     if (this.enabled) {
-      return { enabled: true };
+      return Promise.resolve({ enabled: true });
```

Comment:
as it's an async function I don't need a Promise.resolve here, do I?

Reply:
> The word “async” before a function means one simple thing: a function always returns a promise. Other values are wrapped in a resolved promise automatically. So, async ensures that the function returns a promise, and wraps non-promises in it.

I think we should just return the plain objects in such cases. No need to fiddle around with promises.

Reply:
resolved